python/link_prediction.py

```diff
@@ -0,0 +1,970 @@
+        target_relation: str -> Unique edge type that is used for training.
+        num_neg_per_pos_edge (int): Number of negative edges that will be sampled per one positive edge in the mini-batch.
+        num_layers (int): Number of layers in the GNN architecture.
+        batch_size (int): Batch size used in both training and validation procedure.
+        sampling_workers (int): Number of workers that will cooperate in the sampling procedure in the training and validation.
+
+    Returns:
+        mgp.Record:
+            status (bool): True if parameters were successfully updated, False otherwise.
+            message(str): Additional explanation why method failed or OK otherwise.
+    """
+    global link_prediction_parameters
+
+    _validate_user_parameters(parameters=parameters)
+    
+    for key, value in parameters.items():
+        if not hasattr(link_prediction_parameters, key):
+            return mgp.Record(
+                status=1,
+                message="No attribute " + key + " in class LinkPredictionParameters",
+            )
+        try:
+            setattr(link_prediction_parameters, key, value)
+        except Exception as exception:
+            return mgp.Record(status=1, message=repr(exception))
+
+    # Device type handling
+    if link_prediction_parameters.device_type == CUDA_DEVICE and torch.cuda.is_available() is True:
+        link_prediction_parameters.device_type = CUDA_DEVICE
+    else:
+        link_prediction_parameters.device_type = CPU_DEVICE
+
+    # Lists handling=generator expression + unpacking
+    if type(link_prediction_parameters.hidden_features_size) == tuple:
+        link_prediction_parameters.hidden_features_size = [*(x for x in link_prediction_parameters.hidden_features_size)]
+
+    if type(link_prediction_parameters.attn_num_heads) == tuple:
+        link_prediction_parameters.attn_num_heads = [*(x for x in link_prediction_parameters.attn_num_heads)]
+
+    return mgp.Record(status=1, message="OK")
+
+@mgp.read_proc
+def train(ctx: mgp.ProcCtx,) -> mgp.Record(training_results=mgp.Any, validation_results=mgp.Any):
+    """Train method is used for training the module on the dataset provided with ctx. By taking decision to split the dataset here and not in the separate method, it is impossible to retrain the same model.
+
+    Args:
+        ctx (mgp.ProcCtx, optional): Reference to the process execution.
+
+    Returns:
+        mgp.Record: It returns performance metrics obtained during the training on the training and validation dataset.
+    """
+    # Get global context
+    global training_results, validation_results, predictor, model, graph, reindex_dgl, reindex_orig, features_size_loaded
+
+    # Reset parameters of the old training
+    _reset_train_predict_parameters()
+
+    # Get some data
+    # Dealing with heterogeneous graphs
+    graph, reindex_dgl, reindex_orig = _get_dgl_graph_data(ctx)  # dgl representation of the graph and dict new to old index    
```

Comment:
Instead of having two reindex dictionaries, maybe an idea would be to have one dictionary with two key entries: "dgl" and "orig"/"memgraph". Just dropping ideas